File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    
    def isIsomorphic(self, s: str, t: str) -> bool:
        keys = {}
        values = {}
        
        for i in range(len(s)):
            s_char = s[i]
            t_char = t[i]
            
            if s_char not in keys and t_char not in values:
                keys[s_char] = t_char
                values[t_char] = s_char
            elif s_char in keys and t_char in values:
                if keys[s_char] != t_char or s_char != values[t_char]: return False
            else:
                return False
        
        return True

File: 0205-isomorphic-strings/0205-isomorphic-strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
    const s_t = {} 
    const t_s = {}

    for (let i = 0 ; i < t.length; i++){
        if (!(s[i] in s_t) && !(t[i] in t_s)){
            s_t[s[i]] = t[i]
            t_s[t[i]] = s[i]
        }

        if (!s[i] in s_t || !t[i] in t_s) {
            return false
        }

        if (s_t[s[i]] !== t[i] || t_s[t[i]] !== s[i]){
            return false
        }
    }

    return true
};